test/e2e/network_test.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/retry"
+	wmc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig"
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/nodeconfig"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// hybridOverlayMac is an annotation applied by the hybrid-overlay
+	hybridOverlayMac = "k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac"
+)
+
+// testNetwork runs all the cluster and node network tests
+func testNetwork(t *testing.T) {
+	t.Run("Hybrid overlay annotations", testHybridOverlayAnnotations)
+	t.Run("OpenShift HNS networks", testHNSNetworksCreated)
+}
+
+// testHybridOverlayAnnotations tests that the correct annotations have been added to the bootstrapped node
+func testHybridOverlayAnnotations(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	defer testCtx.cleanup()
+
+	nodes, err := testCtx.kubeclient.CoreV1().Nodes().List(
+		metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	require.NoError(t, err, "unable to get Windows nodes")
+	for _, node := range nodes.Items {
+		assert.NoError(t, testCtx.waitForNodeAnnotation(node.GetName(), nodeconfig.HybridOverlaySubnet))
+		assert.NoError(t, testCtx.waitForNodeAnnotation(node.GetName(), hybridOverlayMac))
+	}
+}
+
+// waitForNodeAnnotation check if the node object has the given annotation and waits for retry.Interval seconds and
+// returns an error if the annotation does not appear in that time frame.
+func (tc *testContext) waitForNodeAnnotation(nodeName, annotation string) error {
+	var found bool
+	err := wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+		node, err := tc.kubeclient.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting node %s: %v", nodeName, err)
+		}
+		_, found := node.Annotations[annotation]
```

Comment:
Please see my comments on this function in the functionality portion of this PR (waitForHybridOverlayAnnotation)

Reply:
Please see my reply there.